mlflow/deployments/interface.py

```diff
@@ -0,0 +1,41 @@
+from mlflow.deployments.plugin_manager import DeploymentPlugins
+
+listType = list
+plugin_store = DeploymentPlugins()
+
+
+# TODO: standardise exceptions
+
+
+def create(target, model_uri, flavor=None, **kwargs):
+    deployment = plugin_store[target].create(model_uri, flavor, **kwargs)
+    if not isinstance(deployment, dict) or \
+            not all([k in ('deployment_id', 'flavor') for k in deployment]):
+        raise TypeError("Deployment creation must return a dictionary with values for "
+                        "``deployment_id`` and ``flavor``")
+    return deployment
+
+
+def delete(target, deployment_id, **kwargs):
+    plugin_store[target].delete(deployment_id, **kwargs)
+
+
+def update(target, deployment_id, model_uri=None, rollback=False, **kwargs):
+    if all((rollback, model_uri)):
+        raise RuntimeError("``update`` has got both ``model_uri`` and ``rollback``")
+    plugin_store[target].update(deployment_id, rollback, model_uri, **kwargs)
+
+
+# TODO: It's a good practise to avoid using ``list`` here.
```

Comment:
Personally, I'd hate to use a reserved keyword here but I could come up with a better name than `list` here. Appreciate any help